adam/language_specific/chinese/chinese_language_generator.py

```diff
@@ -0,0 +1,477 @@
+                for relation in self.situation.always_relations
+                if relation.relation_type == HAS and relation.second_slot == _object
+            ]
+
+            # we can only handle one possession relation at a time right now
+            if len(possession_relations) > 1:
+                raise RuntimeError("Cannot handle multiple possession relations")
+            elif len(possession_relations) == 1:
+                # handle the possession relation if there is one. We don't need to case on person in Chinese
+                # since all possessives are expressed NP+de+NP
+                possessor = self._noun_for_object(possession_relations[0].first_slot)
+                de = DependencyTreeToken("de", PARTICLE)
+                self.dependency_graph.add_edge(de, possessor, role=CASE_POSSESSIVE)
+                self.dependency_graph.add_edge(
+                    possessor, noun_dependency_node, role=NOMINAL_MODIFIER_POSSESSIVE
+                )
+            # if the count is one, we're done since we're not using yi CLF currently
+            if count == 1:
+                return
+            #  https://github.com/isi-vista/adam/issues/782
+            # TODO: get classifiers checked by a native speaker upon implementation
+            elif count == 2:
+                raise NotImplementedError(
+                    "We don't know how to handle Chinese classifiers yet"
+                )
+            # if the count is many, we don't need a CLF
+            else:
+                many = DependencyTreeToken("hen3 dwo1", NUMERAL)
+                self.dependency_graph.add_edge(
+                    many, noun_dependency_node, role=NUMERIC_MODIFIER
+                )
+
+        """Translate relations that the user explicitly calls out"""
+
+        def _translate_relation(
+            self,
+            action: Optional[Action[OntologyNode, SituationObject]],
+            relation: Relation[SituationObject],
+        ):
+            if relation.relation_type == HAS:
+                # if the situation is dynamic, then this will be handled within the NP
+                if (
+                    self.situation.is_dynamic
+                    or IGNORE_HAS_AS_VERB in self.situation.syntax_hints
+                ):
+                    pass
+                else:
+                    raise NotImplementedError
+            elif relation.relation_type == IN_REGION:
+                prepositional_modifier = self.relation_to_prepositional_modifier(
+                    action, relation
+                )
+                if prepositional_modifier:
+                    self.dependency_graph.add_edge(
+                        prepositional_modifier,
+                        self._noun_for_object(relation.first_slot),
+                        role=NOMINAL_MODIFIER,
+                    )
+            else:
+                raise RuntimeError(
+                    "We can't currently translate {} relation to Chinese".format(relation)
```

Comment:
nit: What you have is not wrong at all, but in more recent versions of Python using f-strings is preferred:
```suggestion
                    f"We can't currently translate {relation} relation to Chinese"
```